src/ralph/backends/data/base.py

```diff
@@ -228,6 +296,47 @@ def write(  # pylint: disable=too-many-arguments
                 `ignore_errors` is set to `False`.
             BackendParameterException: If a backend argument value is not valid.
         """
+        if not operation_type:
+            operation_type = self.default_operation_type
+        if operation_type in self.unsupported_operation_types:
+            msg = f"{operation_type.value.capitalize()} operation_type is not allowed."
+            self.logger.error(msg)
+            raise BackendParameterException(msg)
+
+        data = iter(data)
+        try:
+            first_record = next(data)
+        except StopIteration:
+            self.logger.info("Data Iterator is empty; skipping write to target.")
+            return 0
+        data = chain((first_record,), data)
+
+        chunk_size = chunk_size if chunk_size else self.settings.DEFAULT_CHUNK_SIZE
```

Comment:
not linked to this PR in particular but perhaps we could have `DEFAULT_READ_CHUNK_SIZE` & `DEFAULT_WRITE_CHUNK_SIZE` .

Reply:
I think it is a good idea

Reply:
Great Idea) I think we could do this change easily once all backends use this pattern)